### DISCUSSION TITLE
Enbrighten/Jasco 43113

### DISCUSSION THREAD

**Discussion by majorsl** [Koenkk/zigbee2mqtt#13953] [Devices]
Hi all, I'm new as of today, but fairly adept at configuration files for other things.  So, I'm trying to learn how to add this device (and zigbee configurations in general) so I can contribute and submit it.

This Jasco 43113 seems to be an updated version of their 43080 dimmer switch which is supported. It might be as simple as copying the existing 43080 and modifying it to this, but I'm not sure how to get started.  I do see https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html but am not sure where to get the existing 43080 (converter?) file?

It's a bit daunting to a noob.

**Comment by majorsl** [comment]
I've gotten a bit further, but want someone to check my work, if possible.  I grabbed the section of the 43080 from here https://raw.githubusercontent.com/Koenkk/zigbee-herdsman-converters/master/devices/enbrighten.js and modified it with the new model number... that's all I can tell is different.  My external converter file looks like this:

```
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;

const definition = {
    zigbeeModel: ['43113'],
    model: '43113',
    vendor: 'Enbrighten',
    description: 'Zigbee in-wall smart dimmer',
    extend: extend.light_onoff_brightness({noConfigure: true}),
    configure: async (device, coordinatorEndpoint, logger) => {
        await extend.light_onoff_brightness().configure(device, coordinatorEndpoint, logger);
        const endpoint = device.getEndpoint(1);
        await reporting.bind(endpoint, coordinatorEndpoint, ['genOnOff', 'genLevelCtrl']);
        await reporting.onOff(endpoint);
};

module.exports = definition;
```

**Reply by Koenkk** [reply]
This looks good, does it work? If yes make a pr please

**Reply by majorsl** [reply]
> This looks good, does it work? If yes make a pr please

You saw that I did make a PR, thanks for accepting it. Since this is the first time I've done this, what do I do when the version that includes it is released? I assume I delete my own converter file, but do the devices need to be re-paired? I'm cool with that if so.

**Reply by Koenkk** [reply]
You can just delete the external converter file and remove it from your `configuration.yaml`, no repairing needed.
